components/function-controller/internal/webhook/conversion_webhook.go

```diff
@@ -175,22 +175,39 @@ func (w *ConvertingWebhook) convertSpecV1Alpha1ToV1Alpha2(in *serverlessv1alpha1
 		}
 	}
 	out.Spec.Runtime = serverlessv1alpha2.Runtime(in.Spec.Runtime)
-	out.Spec.RuntimeImageOverride = in.Spec.RuntimeImageOverride
+	if in.Spec.RuntimeImageOverride != "" {
+		out.Spec.RuntimeImageOverride = in.Spec.RuntimeImageOverride
+	}
 
 	//TODO: out.Profile
-	//TODO out.CustomRuntimeCOnfiguration
-
-	out.Spec.ResourceConfiguration = serverlessv1alpha2.ResourceConfiguration{
-		Build: serverlessv1alpha2.ResourceRequirements{
-			Resources: in.Spec.BuildResources,
-		},
-		Function: serverlessv1alpha2.ResourceRequirements{
-			Resources: in.Spec.Resources,
-		},
+	//v1BuildEmpty := corev1.ResourceRequirements{}
+	if len(in.Spec.BuildResources.Limits) != 0 || len(in.Spec.BuildResources.Requests) != 0 {
+		if out.Spec.ResourceConfiguration == nil {
+			out.Spec.ResourceConfiguration = &serverlessv1alpha2.ResourceConfiguration{}
+		}
+		if out.Spec.ResourceConfiguration.Build == nil {
+			out.Spec.ResourceConfiguration.Build = &serverlessv1alpha2.ResourceRequirements{}
+		}
+		out.Spec.ResourceConfiguration.Build.Resources = &in.Spec.BuildResources
+	}
+	if len(in.Spec.Resources.Limits) != 0 || len(in.Spec.Resources.Requests) != 0 {
+		if out.Spec.ResourceConfiguration == nil {
+			out.Spec.ResourceConfiguration = &serverlessv1alpha2.ResourceConfiguration{}
+		}
+		if out.Spec.ResourceConfiguration.Function == nil {
+			out.Spec.ResourceConfiguration.Function = &serverlessv1alpha2.ResourceRequirements{}
+		}
+		out.Spec.ResourceConfiguration.Function.Resources = &in.Spec.Resources
```

Comment:
Put it in function `convertResourcesV1Alpha1ToV1Alpha2`

Reply:
Done